pyvrp/tests/test_Statistics.py

```diff
@@ -31,3 +32,27 @@ def test_csv_serialises_correctly(tmp_path):
     # as the previously written object.
     read_stats = Statistics.from_csv(csv_path)
     assert_equal(collected_stats, read_stats)
+
+
+@pytest.mark.parametrize("num_iterations", [0, 1, 10])
+def test_eq(num_iterations: int):
+    data = read("data/OkSmall.txt")
+    cost_evaluator = CostEvaluator(20, 6)
+    rng = RandomNumberGenerator(seed=42)
+    pop = Population(broken_pairs_distance)
+
+    for sol in make_random_solutions(10, data, rng):
+        pop.add(sol, cost_evaluator)
+
+    stats = Statistics()
+    for _ in range(num_iterations):  # populate the statistics object
+        stats.collect_from(pop, cost_evaluator)
+
+    assert_equal(stats, stats)
+    assert_(stats != "test")
+
+    if num_iterations > 0:
+        assert_(stats != Statistics())
+
+        assert_equal(stats.feas_stats[0], stats.feas_stats[0])
+        assert_(stats.feas_stats[0] != "test")
```

Comment:
The `assert_equal`s here are trivial since they test against the same objects right?

Here's my suggestion:

```python
def test_eq():
    data = read("data/OkSmall.txt")
    cost_evaluator = CostEvaluator(20, 6)
    rng = RandomNumberGenerator(seed=42)
    pop = Population(broken_pairs_distance)

    stats = Statistics()
    other = Statistics()

    assert_equal(stats, other)
    assert_(stats != "test")

    # Populate the statistics object.
    for sol in make_random_solutions(10, data, rng):
        pop.add(sol, cost_evaluator)
        stats.collect_from(pop, cost_evaluator)
        other.collect_from(pop, cost_evaluator)

    assert_equal(stats, other)
    assert_(stats != Statistics())
    assert_(stats != "test")

    assert_equal(stats.feas_stats[0], stats.feas_stats[0])
    assert_(stats.feas_stats[0] != stats.feas_stats[1])
    assert_(stats.feas_stats[0] != "test")
```

Reply:
> The `assert_equal`s here are trivial since they test against the same objects right?

No, it is not. This used to fail because of NaN's, which are never equal to themselves. In a sense it's trivial now, but it'd fail again the moment we remove the shortcut in `_Datum.__eq__`.

Reply:
Ah okay, makes much more sense then!

Reply:
I've updated the test a bit to reflect this!